File: python/examples/calc.py
class Calculator(object):
    """A simple, mutable calculator used for testing."""

    def __init__(self):
        self.acc = 0.0

    def __str__(self):
        return '%s' % self.acc
    
    def clr(self):
        self.acc = 0.0

    def add(self,number):
        self.acc += number
        return self.acc

    def div(self,number):
        self.acc /= number
        return self.acc

    def mul(self,number):
        self.acc *= number
        return self.acc

    def sub(self,number):
        self.acc -= number
        return self.acc

    def val(self):
        return self.acc

File: ruby/examples/calc.rb
class Calculator
    # A simple, mutable calculator used for testing.
    # referenced explicitly in ../spec/calculator.spec.rb
    
    def initialize
        @acc = 0.0
    end

    def clr
        @acc = 0.0
    end

    def add(number)
        @acc += number
    end

    def div(number)
        @acc /= number
    end

    def mul(number)
        @acc *= number
    end

    def sub(number)
        @acc -= number
    end

    def val
        return @acc
    end
end
